### PR TITLE
*: add HistogramTimer

### LINKED ISSUE #45: Histogram supports start_timer
Like 

```
let timer = histogram.start_timer();
// do something
timer.observe_duration();
```

### PR DESCRIPTION
**Pull request by overvenus**
Resolved #45

PTAL @siddontang @disksing @brian-brazil

### CODE REVIEW

File: src/histogram.rs [hunk 1]
```diff
@@ -194,6 +194,22 @@ impl Default for HistogramCore {
     }
 }
 
+/// `HistogramTimer` represents an event being timed.
+pub struct HistogramTimer<'a>(&'a Histogram, Instant);
```
Comment (siddontang) [inline src/histogram.rs]: I prefer using field instead of tuple.

File: src/histogram.rs [hunk 2]
```diff
@@ -406,14 +421,16 @@ mod tests {
...
+        thread::sleep(Duration::from_secs(1));
+        timer.observe_duration();
 
         let mf = histogram.collect();
         let m = mf.get_metric().as_ref().get(0).unwrap();
         assert_eq!(m.get_label().len(), 2);
         let proto_histogram = m.get_histogram();
         assert_eq!(proto_histogram.get_sample_count(), 2);
-        assert!((proto_histogram.get_sample_sum() - 1.5).abs() < EPSILON);
+        assert!(proto_histogram.get_sample_sum() > 1.5);
```
Comment (siddontang) [inline src/histogram.rs]: may >= ?

**Comment by siddontang** [comment]
Please upgrade version to 0.2.3

**Comment by siddontang** [comment]
LGTM

**Comment by brian-brazil** [comment]
:+1: This looks about right to me

**Comment by siddontang** [comment]
Thanks @brian-brazil

**Pull request merged**